airflow/providers/openlineage/utils/sql.py

```diff
@@ -0,0 +1,199 @@
+        table_name: str = row[ColumnIndex.TABLE_NAME]
+        table_column: SchemaField = SchemaField(
+            name=row[ColumnIndex.COLUMN_NAME],
+            type=row[ColumnIndex.UDT_NAME],
+            description=None,
+        )
+        ordinal_position = row[ColumnIndex.ORDINAL_POSITION]
+        try:
+            table_database = row[ColumnIndex.DATABASE]
+        except IndexError:
+            table_database = None
+
+        # Attempt to get table schema
+        table_key = ".".join(filter(None, [table_database, table_schema_name, table_name]))
+
+        schemas[table_key] = TableSchema(
+            table=table_name, schema=table_schema_name, database=table_database, fields=[]
+        )
+        columns[table_key].append((ordinal_position, table_column))
+
+    for schema in schemas.values():
+        table_key = ".".join(filter(None, [schema.database, schema.schema, schema.table]))
+        schema.fields = [x for _, x in sorted(columns[table_key])]
+
+    return list(schemas.values())
+
+
+def create_information_schema_query(
+    columns: list[str],
+    information_schema_table_name: str,
+    tables_hierarchy: TablesHierarchy,
+    uppercase_names: bool = False,
+    sqlalchemy_engine: Engine | None = None,
+) -> str:
+    """This function creates query for getting table schemas from information schema."""
+    metadata = MetaData(sqlalchemy_engine)
+    select_statements = []
+    for db, schema_mapping in tables_hierarchy.items():
+        schema, table_name = information_schema_table_name.split(".")
+        if db:
+            schema = f"{db}.{schema}"
+        information_schema_table = Table(
+            table_name, metadata, *[Column(column) for column in columns], schema=schema
+        )
+        filter_clauses = create_filter_clauses(schema_mapping, information_schema_table, uppercase_names)
+        select_statements.append(information_schema_table.select().filter(*filter_clauses))
+    return str(
+        union_all(*select_statements).compile(sqlalchemy_engine, compile_kwargs={"literal_binds": True})
+    )
+
+
+def create_filter_clauses(
+    schema_mapping: dict, information_schema_table: Table, uppercase_names: bool = False
+) -> ClauseElement:
+    """
+    Creates comprehensive filter clauses for all tables in one database (assuming hierarchy
+    of database -> schema -> table).
+
+    :param schema_mapping: a dictionary of schema names and list of tables in each
+    :param information_schema_table: `sqlalchemy.Table` instance used to construct clauses
+        For most SQL dbs it contains `table_name` and `table_schema` columns,
+        therefore it is expected the table has them defined.
+    :param uppercase_names: if True use schema and table names uppercase
+    """
+    filter_clauses = []
+    for schema, tables in schema_mapping.items():
+        filter_clause = information_schema_table.c.table_name.in_(
+            map(lambda name: name.upper() if uppercase_names else name, tables)
```

Comment:
```suggestion
            name.upper() if uppercase_names else name for name in tables
```

Seems shorter and easier to understand to me.

Reply:
Done.